rust/decentraland-godot-lib/src/content/content_provider.rs

```diff
@@ -69,13 +80,62 @@ impl INode for ContentProvider {
             cached: HashMap::new(),
             content_notificator: Arc::new(ContentNotificator::new()),
             godot_single_thread: Arc::new(Semaphore::new(1)),
+            texture_quality: DclConfig::static_get_texture_quality(),
+            tick: 0.0,
         }
     }
     fn ready(&mut self) {}
     fn exit_tree(&mut self) {
         self.cached.clear();
         tracing::info!("ContentProvider::exit_tree");
     }
+
+    fn process(&mut self, dt: f64) {
+        self.tick += dt;
+        if self.tick >= 1.0 {
+            self.tick = 0.0;
+
+            self.cached = self
```

Comment:
Wouldn't it be better to use self.cached.retain() and mutate it instead of creating a new one each time? Even with this rate 1/sec, it makes more sense